### DISCUSSION TITLE
Is there any equivalent to a OpenGL "provoking vertex"?

### DISCUSSION THREAD

**Discussion by AlecGoncharow** [gfx-rs/wgpu#1435] [Q&A] [answered]
Hi,

Apologies in advance if this has been answered elsewhere. 

I am looking into implementing [flat shading](https://www.khronos.org/registry/vulkan/specs/1.0-wsi_extensions/html/chap24.html#vertexpostproc-flatshading) in a project of mine and have come across a few methods to do so. One that piqued my interest was utilizing a method which included clever usage of the [provoking vertex](https://stackoverflow.com/a/60025119) and the `flat` keyword in the glsl code. 

Is this currently possible in wpgu?

**Comment by cwfitzgerald** [comment, marked as answer]
Yup, this will work as expected in wgpu just as it would in OpenGL!

**Reply by AlecGoncharow** [reply]
Thanks for the verification! I left out a `flat` qualifier on my fragment shader input, so I was running into odd behavior.
